internal/middleware/idgate_integration_test.go

```
@@ -0,0 +1,187 @@
+
+func dialReachable(t *testing.T, target string) {
+	t.Helper()
+	u, err := url.Parse(target)
+	if err != nil {
+		t.Fatalf("invalid URL %q: %v", target, err)
+	}
+	host := u.Host
+	if !strings.Contains(host, ":") {
+		host += ":80"
+	}
+	conn, err := net.DialTimeout("tcp", host, 2*time.Second)
```

golangci-lint (2.12.2)

[error] net.DialTimeout must not be called. use (*net.Dialer).DialContext with (*net.Dialer).Timeout

(noctx)